src/components/Profile/PhoneNumberSection.jsx

```diff
@@ -0,0 +1,72 @@
+import React, { useMemo, useState } from 'react'
+
+import { useI18n } from 'cozy-ui/transpiled/react/providers/I18n'
+import { useQuery, useMutation } from 'cozy-client'
+
+import Input from 'components/Input'
+import { buildSettingsInstanceQuery } from 'lib/queries'
+
+const PhoneNumberSection = () => {
+  const { t } = useI18n()
+
+  const { mutate, mutationStatus } = useMutation()
+  const [formError, setFormError] = useState()
+
+  const instanceQuery = buildSettingsInstanceQuery()
+  const { data: instance } = useQuery(
+    instanceQuery.definition,
+    instanceQuery.options
+  )
+
+  const handleBlur = (_, value) => {
+    if (value !== '') {
+      mutate({
+        _rev: instance.meta.rev,
+        ...instance,
+        attributes: {
+          ...instance.attributes,
+          phone_number: value
+        }
+      })
+    }
+  }
+
+  const errors = useMemo(() => {
+    if (mutationStatus === 'failed') {
+      return ['ProfileView.infos.server_error']
+    }
+    if (formError) {
+      return [formError]
+    }
+    return undefined
+  }, [mutationStatus, formError])
+
+  const handleChange = (_, value) => {
+    const phoneRegex = /^[+]?[(]?[0-9]{1,4}[)]?[-\s.]?[0-9]{1,3}[-\s.]?[0-9]{1,4}[-\s.]?[0-9]{1,4}$/
```

Comment:
Can you extract this regex and the `phoneRegex.test` to a function that validate an email and add some unit tests that will also serve as documentation to understand what phone numbers are supported or not.

Reply:
Thank you!